Modules/OrgUnit/classes/Positions/UserAssignment/class.ilOrgUnitUserAssignmentTableGUI.php

```diff
@@ -90,11 +90,18 @@ protected function parseRows($user_ids) {
 	 * @param $user_id
 	 */
 	protected function setRowForUser(&$set, $user_id) {
-		$user = new ilObjUser($user_id);
-		$set["first_name"] = $user->getFirstname();
-		$set["last_name"] = $user->getLastname();
-		$set["user_object"] = $user;
-		$set["user_id"] = $user_id;
+		$ilOrgUnitUserQueries = new ilOrgUnitUserQueries();
+		$user = $ilOrgUnitUserQueries->findUserById($user_id);
```

Comment:
this is a bit confusing, `$ilOrgUnitUserQueries->findUserById($user_id)` will return an array of user data and set is to `$user`, in line 96 `$user` is set to an instance of `ilObjUser`. Please choose other variables for these two things. Additionally `deleteAllAssignmentsOfUser` should not be called from withing the `setRowForUser ` IMO. Please move that to the collection of data in `ilObjOrgUnitTree::getAssignements`as a new method `removeNonExistingUsersFromAssignements`.